ui/management/commands/retranscode_collections.py

```diff
@@ -0,0 +1,271 @@
+"""Management command to schedule retranscoding for collections with multiple filtering options"""
+
+from datetime import datetime
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from django.db.models import Count
+from django.utils.dateparse import parse_datetime
+from django.utils import timezone
+
+from ui.models import Collection
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """Schedule retranscoding for collections based on various filtering criteria"""
+
+    help = __doc__
+
+    def add_arguments(self, parser):
+        # Mutually exclusive group for filtering options
+        filter_group = parser.add_mutually_exclusive_group(required=True)
+
+        filter_group.add_argument(
+            "--ids",
+            nargs="+",
+            type=str,
+            help="Collection IDs (primary keys) to retranscode",
+        )
+
+        filter_group.add_argument(
+            "--all",
+            action="store_true",
+            help="Schedule retranscoding for all collections",
+        )
+
+        filter_group.add_argument(
+            "--owner",
+            type=str,
+            help="Username of the collection owner - retranscode all collections by this user",
+        )
+
+        filter_group.add_argument(
+            "--course-ids",
+            nargs="+",
+            type=str,
+            help="edX course IDs - retranscode all collections with these course IDs",
+        )
```

Comment:
Maybe you can validate like in the snippet below if you want to stick with the space-separated:

```
    def _validate_no_commas(self, value):
        """Custom type function to reject comma-separated values."""
        if "," in value:
            raise ArgumentTypeError(
                f"Invalid value '{value}': Use space-separated values, not comma-separated. "
                f"Example: --arg value1 value2 value3"
            )
        return value
```

```suggestion
        filter_group.add_argument(
            "--course-ids",
            nargs="+",
            type=self._validate_no_commas,
            help="edX course IDs - retranscode all collections with these course IDs",
        )
```